Source/WebCore/platform/ScrollSnapAnimatorState.cpp

```diff
@@ -95,18 +95,72 @@ float ScrollSnapAnimatorState::adjustedScrollDestination(ScrollEventAxis axis, F
-    if (!activeHorizontalIndex || *activeHorizontalIndex >= snapOffsetsForAxis(ScrollEventAxis::Horizontal).size())
-        snapPointChanged |= setNearestScrollSnapIndexForAxisAndOffset(ScrollEventAxis::Horizontal, scrollOffset, scrollExtents, pageScale);
-
     auto activeVerticalIndex = activeSnapIndexForAxis(ScrollEventAxis::Vertical);
-    if (!activeVerticalIndex || *activeVerticalIndex >= snapOffsetsForAxis(ScrollEventAxis::Vertical).size())
+    auto activeHorizontalID = activeSnapIDForAxis(ScrollEventAxis::Horizontal);
+    auto activeVerticalID = activeSnapIDForAxis(ScrollEventAxis::Vertical);
+    auto snapOffsetsVertical = snapOffsetsForAxis(ScrollEventAxis::Vertical);
+    auto snapOffsetsHorizontal = snapOffsetsForAxis(ScrollEventAxis::Horizontal);
+    
+    // Check if we need to set the current indices
+    if (!activeVerticalIndex || *activeVerticalIndex >= snapOffsetsForAxis(ScrollEventAxis::Vertical).size()) {
         snapPointChanged |= setNearestScrollSnapIndexForAxisAndOffset(ScrollEventAxis::Vertical, scrollOffset, scrollExtents, pageScale);
-
+        activeVerticalIndex = activeSnapIndexForAxis(ScrollEventAxis::Vertical);
+    }
+    if (!activeHorizontalIndex || *activeHorizontalIndex >= snapOffsetsForAxis(ScrollEventAxis::Horizontal).size()) {
+        snapPointChanged |= setNearestScrollSnapIndexForAxisAndOffset(ScrollEventAxis::Horizontal, scrollOffset, scrollExtents, pageScale);
+        activeHorizontalIndex = activeSnapIndexForAxis(ScrollEventAxis::Horizontal);
+    }
+    
+    // If we have an active target, see if we need to preserve it
+    if (activeHorizontalID && activeHorizontalIndex) {
+        if (*activeHorizontalID != snapOffsetsHorizontal[*activeHorizontalIndex].snapTargetID)
+            snapPointChanged |= preserveCurrentTargetForAxis(ScrollEventAxis::Horizontal);
+    } else if (activeVerticalID && activeVerticalIndex && *activeVerticalID != snapOffsetsVertical[*activeVerticalIndex].snapTargetID)
+        snapPointChanged |= preserveCurrentTargetForAxis(ScrollEventAxis::Vertical);
+
+    // If we do not have current targets and are snapped to multiple targets, set them
+    if ((!activeHorizontalID && activeHorizontalIndex) && (!activeVerticalID && activeVerticalIndex) && snapOffsetsForAxis(ScrollEventAxis::Horizontal)[*activeHorizontalIndex].snapTargetID != snapOffsetsForAxis(ScrollEventAxis::Vertical)[*activeVerticalIndex].snapTargetID) {
+        setActiveSnapIndexIDForAxis(ScrollEventAxis::Horizontal, snapOffsetsForAxis(ScrollEventAxis::Horizontal)[*activeHorizontalIndex].snapTargetID);
+        setActiveSnapIndexIDForAxis(ScrollEventAxis::Vertical, snapOffsetsForAxis(ScrollEventAxis::Vertical)[*activeVerticalIndex].snapTargetID);
+    }
+    LOG_WITH_STREAM(ScrollSnap, stream << "ScrollSnapAnimatorState::resnapAfterLayout() current target: horizontal" << activeHorizontalID << " vertical: "<< activeVerticalID);
```

Comment:
nit: the logging could be more readable with this:
```suggestion
    LOG_WITH_STREAM(ScrollSnap, stream << "ScrollSnapAnimatorState::resnapAfterLayout() current target: horizontal: " << activeHorizontalID << " vertical: "<< activeVerticalID);
```

Reply:
Fixed.